apis/tempo/v1alpha1/tempomonolithic_types.go

```diff
@@ -0,0 +1,269 @@
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="10Gi"
+	Size resource.Quantity `json:"size"`
+}
+
+// MonolithicTracesStoragePVSpec defines the Persistent Volume configuration.
+type MonolithicTracesStoragePVSpec struct {
+	// Size defines the size of the Persistent Volume for storing the traces. Defaults to 10Gi.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="10Gi"
+	Size resource.Quantity `json:"size"`
+}
+
+// MonolithicIngestionSpec defines the ingestion settings.
+type MonolithicIngestionSpec struct {
+	// OTLP defines the ingestion configuration for OTLP
+	//
+	// +kubebuilder:validation:Optional
+	OTLP *MonolithicIngestionOTLPSpec `json:"otlp,omitempty"`
+
+	// TLS defines the TLS configuration for ingestion
+	//
+	// +kubebuilder:validation:Optional
+	TLS *MonolithicIngestionTLSSpec `json:"tls,omitempty"`
+}
+
+// MonolithicIngestionOTLPSpec defines the settings for OTLP ingestion.
+type MonolithicIngestionOTLPSpec struct {
+	// GRPC defines the OTLP/gRPC configuration
+	//
+	// +kubebuilder:validation:Optional
+	GRPC *MonolithicIngestionOTLPProtocolsGRPCSpec `json:"grpc,omitempty"`
+
+	// HTTP defines the OTLP/HTTP configuration
+	//
+	// +kubebuilder:validation:Optional
+	HTTP *MonolithicIngestionOTLPProtocolsHTTPSpec `json:"http,omitempty"`
+}
+
+// MonolithicIngestionOTLPProtocolsGRPCSpec defines the settings for OTLP ingestion over GRPC.
+type MonolithicIngestionOTLPProtocolsGRPCSpec struct {
+	// Enabled defines if OTLP over gRPC is enabled
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=true
+	Enabled bool `json:"enabled"`
+}
+
+// MonolithicIngestionOTLPProtocolsHTTPSpec defines the settings for OTLP ingestion over HTTP.
+type MonolithicIngestionOTLPProtocolsHTTPSpec struct {
+	// Enabled defines if OTLP over HTTP is enabled
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+}
+
+// MonolithicIngestionTLSSpec defines the TLS settings for ingestion.
+type MonolithicIngestionTLSSpec struct {
+	// Enabled defines if TLS is enabled for ingestion
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+
+	// CA defines the name of a secret containing the CA certificate
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	CA string `json:"ca"`
+
+	// Cert defines the name of a secret containing the TLS certificate and private key
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Cert string `json:"cert"`
+}
+
+// MonolithicJaegerUISpec defines the settings for the Jaeger UI.
+type MonolithicJaegerUISpec struct {
+	// Enabled defines if the Jaeger UI should be enabled
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+
+	// Ingress defines the ingress configuration for Jaeger UI
+	//
+	// +kubebuilder:validation:Optional
+	Ingress *MonolithicJaegerUIIngressSpec `json:"ingress,omitempty"`
```

Comment:
Can be the ingress spec reused from the microservices? There are more settings users might want to configure

Reply:
Already ready in #755:
```
spec:
  jaegerui:                              # JaegerUI defines the Jaeger UI configuration
    enabled: false                       # Enabled defines if the Jaeger UI should be enabled
    ingress:                             # Ingress defines the ingress configuration for Jaeger UI
      annotations:                       # Annotations defines the annotations of the Ingress object.
        "key": ""
      enabled: false                     # Enabled defines if an Ingress object should be created for Jaeger UI
      host: ""                           # Host defines the hostname of the Ingress object.
      ingressClassName: ""               # IngressClassName is the name of an IngressClass cluster resource. Ingress controller implementations use this field to know whether they should be serving this Ingress resource.
    route:                               # Route defines the route configuration for Jaeger UI
      annotations:                       # Annotations defines the annotations of the Ingress object.
        "key": ""
      enabled: false                     # Enabled defines if a Route object should be created for Jaeger UI
      host: ""                           # Host defines the hostname of the Ingress object.
      ingressClassName: ""               # IngressClassName is the name of an IngressClass cluster resource. Ingress controller implementations use this field to know whether they should be serving this Ingress resource.
      termination: "edge"                # Termination specifies the termination type. Default: edge.
```

vs spec of TempoStack:
```
      jaegerQuery:                       # JaegerQuerySpec defines Jaeger Query specific options.
        enabled: false                   # Enabled is used to define if Jaeger Query component should be created.
        ingress:                         # Ingress defines Jaeger Query Ingress options.
          annotations:                   # Annotations defines the annotations of the Ingress object.
            "key": ""
          host: ""                       # Host defines the hostname of the Ingress object.
          ingressClassName: ""           # IngressClassName is the name of an IngressClass cluster resource. Ingress controller implementations use this field to know whether they should be serving this Ingress resource.
          route:                         # Route defines OpenShift Route specific options.
            termination: ""              # Termination specifies the termination type. By default "edge" is used.
          type: ""                       # Type defines the type of Ingress for the Jaeger Query UI. Currently ingress, route and none are supported.
```
(ingressClassName should have been under ingress, as it doesn't apply to route)

Do we value we value _consistency inside the same CR_ or _consistency between the two CRs more_?

Reply:
Same comment, I'd prefer consistency on the same CR